terraform/node_provider.go

```diff
@@ -139,7 +163,29 @@ func (n *NodeApplyableProvider) ConfigureProvider(ctx EvalContext, provider prov
 	}
 
 	configDiags := ctx.ConfigureProvider(n.Addr, unmarkedConfigVal)
+	if configDiags.HasErrors() {
+		if config == nil {
+			// If there isn't an explicit "provider" block in the configuration,
+			// this error message won't be very clear. Add some detail to the
+			// error message in this case.
+			diags = diags.Append(tfdiags.Sourceless(
+				tfdiags.Error,
+				"Invalid provider configuration",
+				fmt.Sprintf(providerConfigErr, configDiags.InConfigBody(configBody).Err(), n.Addr.Provider),
+			))
+			return diags
+		} else {
+			return diags.Append(configDiags.InConfigBody(configBody))
+		}
+	}
 	diags = diags.Append(configDiags.InConfigBody(configBody))
 
 	return diags
 }
+
+const providerConfigErr = `%s
+
+Provider %q requires explicit configuration. Add a provider block to the root
+module and configure the provider's required arguments as described in the
+provider documentation.
+`
```

Comment:
Writing this paragraph with embedded newlines has the unfortunate effect of causing odd line-wrapping in the diagnostic output. You can see this in the screenshot below: it wraps awkwardly between "root" and "module" when there's no need to:

![image](https://user-images.githubusercontent.com/68917/101917062-c965f400-3b95-11eb-827b-a77110d55475.png)

Reply:
I already tried changing the newlines - including one long single line with no newlines - and it didn't improve the output, do you know how else I might improve this?

Reply:
Huh. Making the last paragraph a single line seemed to work for me:

```patch
diff --git a/terraform/node_provider.go b/terraform/node_provider.go
index 57245fafb..001408bf2 100644
--- a/terraform/node_provider.go
+++ b/terraform/node_provider.go
@@ -185,7 +185,4 @@ func (n *NodeApplyableProvider) ConfigureProvider(ctx EvalContext, provider prov

 const providerConfigErr = `%s

-Provider %q requires explicit configuration. Add a provider block to the root
-module and configure the provider's required arguments as described in the
-provider documentation.
-`
+Provider %q requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in the provider documentation.`
```

Result:

![image](https://user-images.githubusercontent.com/68917/101926140-cae8e980-3ba0-11eb-99f7-29c26bea996f.png)

Reply:
I apologize, my error - I tried what I _thought_ I already tried before and it now looks better. Sorry!